Added `add_ess_header()` function

ddsjoberg opened the pull request:
PR adds the basics of what you'll need to replace the default headers in `tbl_svysummary()` with ESS-calculated numbers.

This does add new dependencies, some of which you could do a little re-writing to remove.
FYI to run the examples in the help file, the user needs survey, gtsummary, cards, and cardx packages, which is why all of them have been added to suggests. I used `gtsummary::trial` data set in the examples, but I am sure you have a more suitable example on hand, so feel free to update.

The function still needs unit tests, but I thought you could write them after you confirm I didn't implement the ESS calculation incorrectly 😜 

I also ran `usethis::use_github_action("test-coverage.yaml")` because the v3 of the artifact upload workflow is now deprecated.

closes #11

malcolmbarrett commented:
Thanks, @ddsjoberg! I added some tests and tweaked the docs a bit. 

One thing I changed was to label it as ESS. Is there a good way to make an automatic footnote to explain the acronym?

ddsjoberg commented on R/add_ess_header.R:
```diff
@@ -0,0 +1,89 @@
+#'
+#' gtsummary::tbl_svysummary(svy, by = qsmk, include = c(age, sex, smokeyrs)) |>
+#'   add_ess_header(header = "**{level}**  \nN = {n_unweighted}; \nESS = {round(n, 1)}")
+add_ess_header <- function(x, header = "**{level}**  \nESS = {round(n, 1)}") {
```
If you'd like the default to round the ESS to one decimal place, I suggest using `format()` because if a number rounds to an integer, it won't show the `0` in the decimal place, e.g. `40` instead of `40.0`. You could also use `gtsummary::style_numbe(n, digits = 1)`.

malcolmbarrett replied:
Good point. Is the base equivalent to the `style_number()` example roughly `format(round(n, 1), nsmall = 1)`? Seems like at that point I might as well use `style_number()`

malcolmbarrett replied:
Oh I guess I don't need `round()`: `format(n, digits = 1, nsmall = 1)`

ddsjoberg commented:
> One thing I changed was to label it as ESS. Is there a good way to make an automatic footnote to explain the acronym?

Yes! In the next release (hoping to go to CRAN in the next 3 weeks or so), there is a new function just for abbreviations.


```r
svy <- survey::svydesign(~1, data = nhefs_weights, weights = ~ w_ate)
gtsummary::tbl_svysummary(svy, include = c(age, sex, smokeyrs)) |>
  add_ess_header() |> 
  gtsummary::modify_abbreviation("ESS = Estimate Sample Size")
```

![image](https://github.com/user-attachments/assets/62f7f8ce-f65e-4572-90ac-fecdd4c172c6)

We could add a line at the end of the `add_ess_header()` function like this:

```r
if (grepl(pattern = "ESS", x =header, fixed = TRUE)) {
  x <- gtsummary::modify_abbreviation(x, "ESS = Estimate Sample Size")
}
```

ddsjoberg commented:
Ahhh, lewking cute!

![image](https://github.com/user-attachments/assets/c93ace5f-ffb7-4671-8236-d81bcace41e3)

malcolmbarrett commented:
Yaaa! Almost there, just fixing some R CMD Check issues.

Incidentally, I have `warnPartialMatchArgs = TRUE` turned on, and I see:

```r
Warning messages:
1: In cards::nest_for_ard(data$variables, by = by, key = "...ard_no_one_will_ever_pick_this...",  :
  partial argument match of 'rename' to 'rename_columns'
```

I don't think this stems from this change so I thought I'd point it out to you

malcolmbarrett commented:
Thanks a million, @ddsjoberg! This is awesome

ddsjoberg commented:
Ah, yes, thanks for pointing out the partial match. I'll update (someday...)

malcolmbarrett commented:
@ddsjoberg do you think it would be easy extend this to the overall column, e.g., `add_ess_overall()`?

ddsjoberg commented:
> @ddsjoberg do you think it would be easy extend this to the overall column, e.g., `add_ess_overall()`?

yes! i forgot to add that, oops! i'll add it to the primary function to handle this case automatically
